src/ayb_db/db_interfaces.rs

```diff
@@ -262,34 +262,63 @@ WHERE id = $1
                 Ok(entity)
             }
 
-            async fn update_entity_by_id(&self, entity: &Entity, entity_id: i32) -> Result<InstantiatedEntity, AybError> {
-                let entity: InstantiatedEntity = sqlx::query_as(
-                    r#"
-UPDATE entity SET
-    description = $1,
-    organization = $2,
-    location = $3,
-    display_name = $4,
-    links = $5
-WHERE entity.id = $6
-RETURNING id, slug, entity_type, display_name, description, organization, location, links
-                    "#
-                )
-                .bind(&entity.description)
-                .bind(&entity.organization)
-                .bind(&entity.location)
-                .bind(&entity.display_name)
-                .bind(serde_json::to_value(&entity.links)?)
-                .bind(entity_id)
-                .fetch_one(&self.pool)
-                .await
-                .or_else(|err| match err {
-                    sqlx::Error::RowNotFound => Err(AybError::RecordNotFound {
-                        id: entity_id.to_string(),
-                        record_type: "entity".into(),
-                    }),
-                    _ => Err(AybError::from(err)),
-                })?;
+            async fn update_entity_by_id(&self, entity: &PartialEntity, entity_id: i32) -> Result<InstantiatedEntity, AybError> {
+                let mut query = QueryBuilder::new("UPDATE entity SET");
+                if let Some(description) = &entity.description {
```

Comment:
I bet we can shorten the repetition on the next few branches by creating a vec/array of pairs like `("description", entity.description)` for any non-`None` item, and then looping over it.

Reply:
That's a good idea :)